### ISSUE TITLE
`F.logsigmoid(input, out=blah)` crashes

### ISSUE THREAD

**Issue by zou3519** [pytorch/pytorch#36499]
## 🐛 Bug

On master,
```
import torch
import torch.nn.functional as F
x = torch.randn(3, 3)
y = torch.randn(9)
F.logsigmoid(x, out=y) # crashes
```

The crash does not occur on CUDA on master.

However, in v1.4.0, the above snippet does not crash on CPU or CUDA.

## Expected behavior

Either:
- No crash
- F.logsigmoid should not take an `out` argument.  The docs imply there is no out argument: https://pytorch.org/docs/stable/nn.functional.html#torch.nn.functional.logsigmoid

**Comment by kshitij12345** [comment]
Apparently there is a signature of log_sigmoid which takes out argument in the `native_functions.yaml` thus the keyword argument is allowed. However, not sure if that variant is used anywhere else.
https://github.com/pytorch/pytorch/blob/765bf8f03ddcf74fcaf299ea3f8577a2e254e1a7/aten/src/ATen/native/native_functions.yaml#L5799-L5805

**Comment by vadimkantorov** [comment]
(about naming: at some point in the past I suggested that `log_softmax` and `log_sigmoid` used underscores consistently)

**Comment by zou3519** [comment]
@kshitij12345 If we ultimately decide to remove the out variant due to it not being in the documentation, we should probably still restore the behavior and perform a deprecation cycle (deprecate the out keyword arg in a release, then remove it entirely in the next release).

**Comment by kshitij12345** [comment]
I am confused as to what the deprecation warning or note will be for the undocumented argument? :thinking:

**Comment by zou3519** [comment]
I'm looking at the problem from an overly conservative perspective, but to answer your question, something like "We've deprecated the out argument for logsigmoid" should be sufficient. Only people who use it will stumble upon it.

**Comment by vadimkantorov** [comment]
(there's also a possibility of supporting it properly at some point in future. if we say "deprecated", it suggests that it won't come back. maybe better sth like "is not supported for now"?)

**Comment by kshitij12345** [comment]
@vadimkantorov I don't think the one exposed in functional should have `out`.

**Comment by vadimkantorov** [comment]
Well, it may be nice to have `out`-variants even for fused functions like `log_sigmoid`. If I understand well, originally `nn.functional` existed to make sure that functions there worked with variables and were differentiable. But now that variables and tensors are merged, for many functions this distinction/mention doesn't make sense anymore IMO. PyTorch is now used in a more general scientific computing context, not just for training neural nets. So things like F.normalize, F.logsumexp, F.softmax, F.log_softmax, F.logsigmoid are very useful in other contexts as well, and it may be useful to have `out` variant (for now `out` differentiation is not supported, but it was discussed at some point and was decided to support it in future for some limited cases IIRC) or at least inplace variant. Personally, I think that it makes sense to officially move many of these functions to `torch.` and to `tensor.*` instance methods, since they are so common.

It's understandable that they are not all supported for now, but if the placement of a function in `functional` is a destiny, it's unfortunate as a reason for absence inplace and (more general) out-variants for memory-economic computations.

**Comment by gchanan** [comment]
works properly on cuda.
